app/models/reading.rb

```diff
@@ -2,7 +2,7 @@ class Reading < ActiveRecord::Base
   belongs_to :user
   belongs_to :work
 
-  after_save :expire_cached_home_marked_for_later, if: :toread?
+  after_save :expire_cached_home_marked_for_later, if: :toread_changed?
   after_destroy :expire_cached_home_marked_for_later, if: :toread?
```

Comment:
@sarken, would you also need to use [code]:toread_changed?[/code] for the [code]:after_destroy[/code] scope as well?